frontend/src/pages/Projects/Project/ProjectDetails/index.tsx

```diff
@@ -0,0 +1,54 @@
+import { LinearProgress } from "@mui/material";
+import React from "react";
+import { useQuery } from "react-query";
+import { useParams } from "react-router-dom";
+import { GDX_Accordion } from "../../../../components/GDX_Accordion";
+import { apiAxios } from "../../../../utils";
+import { AgreementSection } from "./AgreementSection";
+import { ProjectRegistrationSection } from "./ProjectRegistrationSection";
+
+export const ProjectDetails = () => {
+  const { projectId } = useParams();
+
+  const getProject = async () => {
+    const project = await apiAxios().get(`projects/${projectId}`);
+    return project;
+  };
+
+  // Queries
+  const projectQuery = useQuery(`project - ${projectId}`, getProject, {
+    refetchOnWindowFocus: false,
+    retryOnMount: false,
+    refetchOnReconnect: false,
+    retry: false,
+    staleTime: Infinity,
+  });
+  return (
+    <>
+      {(() => {
```

Comment:
Why is this an IIFE?

Reply:
This is the only way I could get switch statements to work to render react components conditionally.